dango/types/src/vesting.rs

```diff
@@ -0,0 +1,139 @@
+use {
+    anyhow::ensure,
+    grug::{
+        Addr, Coin, Duration, MultiplyFraction, NumberConst, Timestamp, Udec128, Uint128, Undefined,
+    },
+    std::collections::BTreeMap,
+};
+
+pub type ClaimablePosition = Position<Uint128>;
+
+#[grug::derive(Serde)]
+pub struct InstantiateMsg {}
+
+#[grug::derive(Serde)]
+pub enum ExecuteMsg {
+    CreatePosition {
+        user: Addr,
+        schedule: Schedule<Option<Timestamp>>,
+    },
+    Claim {
+        idx: u64,
+    },
+}
+
+#[grug::derive(Serde, QueryRequest)]
+pub enum QueryMsg {
+    #[returns(ClaimablePosition)]
+    Position { idx: u64 },
```

Comment:
I would create a `PositionIndex` alias, similar to `AccountIndex` in dango-account-factory.